src/collectors/windows.plugin/perflib-hyperv.c

```diff
@@ -0,0 +1,1553 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "windows_plugin.h"
+#include "windows-internals.h"
+
+#define _COMMON_PLUGIN_NAME "windows.plugin"
+#define _COMMON_PLUGIN_MODULE_NAME "PerflibHyperV"
+#include "../common-contexts/common-contexts.h"
+
+#define HYPERV  "hyperv"
+
+
+long chart_priority = NETDATA_CHART_PRIO_WINDOWS_HYPERV;
+
+void get_and_sanitize_instance_value(PERF_DATA_BLOCK *pDataBlock,  PERF_OBJECT_TYPE *pObjectType, PERF_INSTANCE_DEFINITION *pi, char *buffer, size_t buffer_size)
+{
+    char wstr[8192];
+    if (!getInstanceName(pDataBlock, pObjectType, pi, wstr, sizeof(wstr))) {
+        strncpyz(buffer, "[unknown]", buffer_size - 1);
+        return;
+    }
+    rrdlabels_sanitize_value(buffer, wstr, buffer_size);
+}
+
+#define DICT_PERF_OPTION (DICT_OPTION_DONT_OVERWRITE_VALUE | DICT_OPTION_FIXED_SIZE)
+
+#define DEFINE_RD(counter_name) RRDDIM *rd_##counter_name
+
```

Comment:
We could also do this with the following macro, appending a prefix like `ND_PERFLIB`.

Reply:
What do you mean?

Reply:
I thought name `DEFINE_RD` too generic.  So my suggestion was do rename it to `ND_DEFINE_RD`, for example.